executor/admin.go

```diff
@@ -457,3 +459,165 @@ func (e *RecoverIndexExec) NextChunk(ctx context.Context, chk *chunk.Chunk) erro
 	e.done = true
 	return nil
 }
+
+// CleanupIndexExec represents a cleanup index executor.
+// It is built from "admin cleanup index" statement, is used to delete
+// dangling index data.
+type CleanupIndexExec struct {
+	baseExecutor
+
+	done bool
+
+	index     table.Index
+	table     table.Table
+	batchSize uint64
+
+	idxCols          []*model.ColumnInfo
+	idxColFieldTypes []*types.FieldType
+	idxChunk         *chunk.Chunk
+	idxResult        distsql.SelectResult
+
+	//tblChunk         *chunk.Chunk
+	//tblResult        distsql.SelectResult
+
+	matchedIndex     map[int64]idxData
+}
+
+type idxData struct {
+	matched bool
+	idxVals []types.Datum
+}
+
+func (e *CleanupIndexExec) getIdxColTypes() []*types.FieldType {
+	if e.idxColFieldTypes != nil {
+		return e.idxColFieldTypes
+	}
+	e.idxColFieldTypes = make([]*types.FieldType, 0, len(e.idxCols))
+	for _, col := range e.idxCols {
+		e.idxColFieldTypes = append(e.idxColFieldTypes, &col.FieldType)
+	}
+	return e.idxColFieldTypes
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (e *CleanupIndexExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.done {
+		return nil
+	}
+	handleIdx := len(e.idxCols) - 1
+	errInTxn := kv.RunInNewTxn(e.ctx.GetStore(), true, func(txn kv.Transaction) error {
```

Comment:
Do we need to handle this code in batch? I'm afraid that we need to clean up a lot of data. At the same time, some people also modified the data. Then we need to redo it.